Mathlib/Topology/Irreducible.lean

```diff
@@ -311,4 +313,79 @@ theorem IsPreirreducible.preimage (ht : IsPreirreducible t) {f : Y → X}
   cases hf.injective h₄
   exact ⟨y, h₁, h₂, h₃⟩
 
+section
+
+open Set.Notation
+
+@[stacks 004Z]
+lemma IsPreirreducible.preimage_of_dense_isPreirreducible_fiber
```

Comment:
I wasn't not sure we should keep `dense` in the name given that we dropped it from the statement. OTOH this not such an easy lemma to name, and the alternatives are either very long or more opaque so I think we can leave the name as it stands.